firefly/kernel/init/init.cpp

```diff
@@ -98,29 +97,27 @@ void* stivale2_get_tag(stivale2_struct* stivale2_struct, uint64_t id)
 }
 
 using namespace firefly::kernel;
-mm::pmm::Freelist<mm::pmm::PhysicalAddress> mm::pmm::freelist;
-
-void bootloader_services_init(stivale2_struct* handover)
-{
+void bootloader_services_init(stivale2_struct* handover) {
     auto tag_term = static_cast<stivale2_struct_tag_terminal*>(stivale2_get_tag(handover, STIVALE2_STRUCT_TAG_TERMINAL_ID));
```

Comment:
This is what i mean, if it returns a tag_terminal then return that?

Reply:
We can't simply return a struct ourselves, this is something the bootloader provides and we need to check if it exists.
It'll fill out all the fields in the response as well

Reply:
Ight just wanted you to consider it. All g